tracing/dummy_span.go

```diff
@@ -15,6 +16,8 @@ import (
 // Any child spans created from a DummySpan will be linked to the dummy's
 // next non-dummy ancestor.
 type DummySpan struct {
+	embedded.Span
```

Comment:
TL;DR for the embedded.Span add?

Reply:
Since otel v1.20.0, they add this as a version specific placeholder.  From the source:

https://github.com/open-telemetry/opentelemetry-go/blob/trace/v1.21.0/trace/embedded/embedded.go#L48-L56

```go
// Span is embedded in [go.opentelemetry.io/otel/trace.Span].
//
// Embed this interface in your implementation of the
// [go.opentelemetry.io/otel/trace.Span] if you want users to experience a
// compilation error, signaling they need to update to your latest
// implementation, when the [go.opentelemetry.io/otel/trace.Span] interface is
// extended (which is something that can happen without a major version bump of
// the API package).
```

Apparently, the OTel devs want a compile error on legacy implementations of `Span` because they put `embedded.Span` in the main `Span` interface.  There's no code, it's just a placeholder function that acts like a C++ class trait.  It's not clear what the strategy is behind this other than break our build.